### PR TITLE
LP-2692 - Build a mechanism to support Business Line prerequisite courses

### PR DESCRIPTION
**Pull request by MunamMubashir-arbisoft**
[LP-2692](https://philanthropyu.atlassian.net/browse/LP-2692)

- Update MultilingualCourseGroup model to allow common, program and specific Business Line prereqs
- Add Admin for MultilingualCourseGroup
- Add MultilingualCourseGroupForm to allow prereq related validations

### CODE REVIEW

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/1879/workflows/workflows_id/jobs/1879/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/admin.py (50.0%): Missing lines 229-234
openedx/adg/lms/applications/constants.py (100%)
openedx/adg/lms/applications/forms.py (100%)
openedx/adg/lms/applications/managers.py (100%)
openedx/adg/lms/applications/models.py (100%)
openedx/adg/lms/applications/tests/factories.py (100%)
openedx/adg/lms/applications/tests/test_forms.py (100%)
openedx/adg/lms/applications/tests/test_models.py (100%)
-------------
Total:   53 lines
Missing: 6 lines
Coverage: 88%
-------------
```

File: openedx/adg/lms/applications/models.py [hunk 1]
```diff
@@ -361,6 +378,10 @@ class Meta:
     def __str__(self):
         return self.name
 
+    @property
+    def is_business_line_prerequisite(self):
+        return self.business_line_prerequisite is not None
```
Comment (ahmed-faraz46) [inline openedx/adg/lms/applications/models.py:383]: How about `bool(self.business_line_prerequisite)`?

File: openedx/adg/lms/applications/tests/test_forms.py [hunk 2]
```diff
@@ -228,3 +229,34 @@ def test_user_application_cover_letter_form_file_size(size, expected, mock_cover
...
+    """
+    form_data = {
+        'name': 'test',
+        'is_program_prerequisite': is_program_prereq,
+        'is_common_business_line_prerequisite': is_common_for_all_prereq,
+        'business_line_prerequisite': BusinessLineFactory().id if is_business_line_prereq else None
+    }
+
+    form = MultilingualCourseGroupForm(data=form_data)
+    assert form.is_valid() is expected_result
```
Comment (ahmed-faraz46) [inline openedx/adg/lms/applications/tests/test_forms.py:262]: We should also verify if the error message is associated with correct field?

**Review by ahmed-faraz46 — CHANGES_REQUESTED** [review]
Good work Munam 👏 Here are a few suggestions please look into them.

File: openedx/adg/lms/applications/admin.py [hunk 3]
```diff
@@ -186,8 +186,52 @@ def get_extra(self, request, obj=None, **kwargs):
...
+        Extend the `get_form` method of `MultilingualCourseGroupAdmin` to remove the widgets that allow the option to
+        add, change and remove Business Lines from the MultilingualCourseGroup admin model page.
+
+        Arguments:
+            request (WSGIRequest): HTTP request for the `MultilingualCourseGroupAdmin` page
+            obj (MultilingualCourseGroup): `MultilingualCourseGroup` object
+            change (bool): Type of form
+
+        Returns:
+            MultilingualCourseGroupForm: `MultilingualCourseGroupForm` with updated permissions rearding related
```
Comment (annas-usman) [inline openedx/adg/lms/applications/admin.py:226]: regarding*

**Review by annas-usman — COMMENTED** [review]
Apart from the already requested changes everything looks good, just found a spelling mistake.

File: openedx/adg/lms/applications/constants.py [hunk 4]
```diff
@@ -70,3 +70,7 @@
 HTML_FOR_EMBEDDED_FILE_VIEW = '<iframe src="{path_to_file}" style="width:889px; height:393px;"></iframe>'
 
 ADG_ADMIN_GROUP_NAME = 'ADG Admins'
+
+COURSE_GROUP_PREREQ_VALIDATION_ERROR = _(
+    'A Course Group can either be a prerequisite for the Program, specific Business Line or common for all'
```
[the remark this answers is not in the document]
Comment (javeriaejaz-arbisoft) [reply openedx/adg/lms/applications/constants.py:75]: The error message seems a bit ambiguous. Can we please change it along the line `A course group can only be a prerequisite for one of the following: ......` ?

**Review by javeriaejaz-arbisoft — COMMENTED** [review]
Just added a comment for error message. The rest looks fine, though we can improve the coverage by writing the test case for admin.

File: openedx/adg/lms/applications/constants.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (MunamMubashir-arbisoft) [reply openedx/adg/lms/applications/constants.py:75]: @javeriaejaz-arbisoft I have already added a similar message for the fieldset description.

**Comment by MunamMubashir-arbisoft** [comment]
@ahmed-faraz46 I've added a custom migration. On my local deploy, all the data is being preserved with this. Please take a look.

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/1888/workflows/workflows_id/jobs/1888/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/admin.py (93.3%): Missing lines 216
openedx/adg/lms/applications/constants.py (100%)
openedx/adg/lms/applications/forms.py (100%)
openedx/adg/lms/applications/managers.py (100%)
openedx/adg/lms/applications/models.py (100%)
openedx/adg/lms/applications/tests/conftest.py (100%)
openedx/adg/lms/applications/tests/factories.py (100%)
openedx/adg/lms/applications/tests/test_admin/test_multilingual_course_group_admin.py (100%)
openedx/adg/lms/applications/tests/test_forms.py (100%)
openedx/adg/lms/applications/tests/test_models.py (100%)
-------------
Total:   75 lines
Missing: 1 line
Coverage: 98%
-------------
```

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/1889/workflows/workflows_id/jobs/1889/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/admin.py (93.3%): Missing lines 216
openedx/adg/lms/applications/constants.py (100%)
openedx/adg/lms/applications/forms.py (100%)
openedx/adg/lms/applications/managers.py (100%)
openedx/adg/lms/applications/models.py (100%)
openedx/adg/lms/applications/tests/conftest.py (100%)
openedx/adg/lms/applications/tests/factories.py (100%)
openedx/adg/lms/applications/tests/test_admin/test_multilingual_course_group_admin.py (100%)
openedx/adg/lms/applications/tests/test_forms.py (100%)
openedx/adg/lms/applications/tests/test_models.py (100%)
-------------
Total:   75 lines
Missing: 1 line
Coverage: 98%
-------------
```

**Review by annas-usman — APPROVED** [review]

**Pull request merged**